base/deepcopy.jl

```diff
@@ -25,7 +25,10 @@ so far within the recursion. Within the definition, `deepcopy_internal` should b
 in place of `deepcopy`, and the `dict` variable should be
 updated as appropriate before returning.
 """
-deepcopy(x) = deepcopy_internal(x, IdDict())::typeof(x)
+function deepcopy(x)
+    isbitstype(typeof(x)) && return x
```

Comment:
Could this just be `isbits(x)`?

Reply:
Morally, that should be possible. But the compiler doesn't pick up on that:
```julia
julia> struct Foo; x::Int; end

julia> function fastdeepcopy(x)
           isbits(x) && return x
           return deepcopy_internal(x, IdDict())::typeof(x)
       end
fastdeepcopy (generic function with 1 method)

julia> fastdeepcopy(Foo(2))
Foo(2)

julia> @code_warntype fastdeepcopy(Foo(2))
Variables
  #self#::Core.Compiler.Const(fastdeepcopy, false)
  x::Foo

Body::Foo
1 ─ %1 = Main.isbits(x)::Bool
└──      goto #4 if not %1
2 ─      return x
3 ─      Core.Compiler.Const(:(goto %6), false)
4 ┄      false
│   %6 = Main.IdDict()::IdDict{Any,Any}
│   %7 = Main.deepcopy_internal(x, %6)::Any
│   %8 = Main.typeof(x)::Core.Compiler.Const(Foo, false)
│   %9 = Core.typeassert(%7, %8)::Foo
└──      return %9

```

Reply:
Hitting "resolve" on this one.